File: party_planner/lib/enums/expense_status.dart
import 'package:flutter/material.dart';

// ignore_for_file: constant_identifier_names

enum ExpenseStatus {
  pending,
  approved_by_management,
  paid,
}

extension ExpenseStatusExtension on ExpenseStatus {
  String get description {
    switch (this) {
      case ExpenseStatus.pending:
        return 'قيد الانتظار';
      case ExpenseStatus.approved_by_management:
        return 'موافقة الإدارة';
      case ExpenseStatus.paid:
        return 'مدفوع';
    }
  }

  IconData get icon {
    switch (this) {
      case ExpenseStatus.pending:
        return Icons.hourglass_empty; // رمز الانتظار
      case ExpenseStatus.approved_by_management:
        return Icons.check_circle_outline; // رمز الموافقة
      case ExpenseStatus.paid:
        return Icons.payment; // رمز الدفع
    }
  }

  Color get color {
    switch (this) {
      case ExpenseStatus.pending:
        return Colors.orange; // اللون البرتقالي
      case ExpenseStatus.approved_by_management:
        return Colors.green; // اللون الأخضر
      case ExpenseStatus.paid:
        return Colors.blue; // اللون الأزرق
    }
  }
}

File: backend/app/Enums/ExpenseStatus.php
<?php

namespace App\Enums;

use BenSampo\Enum\Enum;

final class ExpenseStatus extends Enum
{
    const PENDING = 'pending';
    const APPROVED_BY_MANAGEMENT = 'approved_by_management';
    const PAID = 'paid';

    public static function getDescription($value): string
    {
        switch ($value) {
            case self::PENDING:
                return 'قيد الانتظار';
            case self::APPROVED_BY_MANAGEMENT:
                return 'موافقة الإدارة';
            case self::PAID:
                return 'مدفوع';
            default:
                return self::getKey($value);
        }
    }
}
